CMakeLists.txt

```diff
@@ -442,7 +442,7 @@ if(${VELOX_BUILD_MINIMAL_WITH_DWIO}
 
   # Locate or build protobuf.
   set_source(Protobuf)
-  resolve_dependency(Protobuf 3.21.7...<4)
+  resolve_dependency(Protobuf CONFIG REQUIRED)
```

Comment:
Do we really want to drop the lower bound too?

Reply:
Yes, when protobuf is preinstalled, it reports version 5.27.2.
However, when it is used as cmake module it reports version 27.2.
The same happen happens with the current version (21.8 vs 3.21.8).
It also complicates versioning the dependencies, because we need to keep track of every cmake file that refers to a version. I may put back the explicit versions if you disagree.

Reply:
We should keep the minimum `3.21.7 (21.7)`. I wonder if there is a way to normalize the version numbers.

Reply:
The official findprotobuf.cmake has pretty extensive logic for the versions, maybe that can handle the different version schemas? (E.g. add `MODULE`) https://github.com/Kitware/CMake/blob/master/Modules%2FFindProtobuf.cmake#L590

Reply:
Agree. I am investigating this.

Reply:
Also we only care about the pre installed version here as our bundled version is fixed and not checked.

Reply:
Fixed. I had not realized that by using CONFIG in the find_package, we always get the version that has the ABI as prefix, 3.21.7 in this case. I also removed the <4 because the newest protobuf is now in ABI version 5.